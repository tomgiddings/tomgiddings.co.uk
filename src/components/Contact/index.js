// @flow
import React from 'react';
import Footer from '../Footer'

class Contact extends React.Component<*> {
  constructor(props) {
    super(props);
    this.submitForm = this.submitForm.bind(this);
    this.state = {
      status: ""
    };
  }
  submitForm(ev) {
    ev.preventDefault();
    const form = ev.target;
    const data = new FormData(form);
    const xhr = new XMLHttpRequest();
    xhr.open(form.method, form.action);
    xhr.setRequestHeader("Accept", "application/json");
    xhr.onreadystatechange = () => {
      if (xhr.readyState !== XMLHttpRequest.DONE) return;
      if (xhr.status === 200) {
        form.reset();
        this.setState({ status: "SUCCESS" });
      } else {
        this.setState({ status: "ERROR" });
      }
    };
    xhr.send(data);
  }
  render() {
    const { status } = this.state;
    document.title = 'Tom Giddings - Contact';
    return (
      <article className="page contact">
        <div className="content-wrapper">
          <h1>Contact Me</h1>
          <form onSubmit={this.submitForm} action="https://formspree.io/xknolrqo" method="POST">
            <label>Email:</label>
            <input type="email" name="email" />
            <label>Message:</label>
            <textarea name="message" rows="5" required></textarea>
            {status === "SUCCESS" ? <p>Thankyou! I'll get back to you as soon as possible.</p> : <button>Send</button>}
            {status === "ERROR" && <p>Ooops! There was an error. Please try again.</p>}
          </form>
          <ul>
            <li><a href="https://github.com/tomgiddings"><i className="fab fa-4x fa-github-square"></i></a></li>
            <li><a href="https://www.linkedin.com/in/tomgiddings/"><i className="fab fa-4x fa-linkedin"></i></a></li>
            <li><a href="https://instagram.com/tomgiddings/"><i className="fab fa-4x fa-instagram"></i></a></li>
          </ul>
        </div>
      </article>
    );
  }
}
export default Contact;
