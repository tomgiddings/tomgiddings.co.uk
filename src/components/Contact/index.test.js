import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Contact from './';

afterEach(cleanup);

it('Renders correctly', () => {
  const { asFragment } = render(<Contact />);
  expect(asFragment()).toMatchSnapshot();
});
