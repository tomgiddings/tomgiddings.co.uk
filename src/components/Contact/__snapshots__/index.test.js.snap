// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<DocumentFragment>
  <article
    class="page contact"
  >
    <div
      class="content-wrapper"
    >
      <h1>
        Contact Me
      </h1>
      <form
        action="https://formspree.io/xknolrqo"
        method="POST"
      >
        <label>
          Email:
        </label>
        <input
          name="email"
          type="email"
        />
        <label>
          Message:
        </label>
        <textarea
          name="message"
          required=""
          rows="7"
        />
        <button>
          Send
        </button>
      </form>
      <h4>
        Alternatively, you can contact me via the following:
      </h4>
      <ul>
        <li>
          <a
            href="https://github.com/tomgiddings"
          >
            <i
              class="fab fa-4x fa-github-square"
            />
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com/in/tomgiddings/"
          >
            <i
              class="fab fa-4x fa-linkedin"
            />
          </a>
        </li>
        <li>
          <a
            href="https://instagram.com/tomgiddings/"
          >
            <i
              class="fab fa-4x fa-instagram"
            />
          </a>
        </li>
      </ul>
    </div>
  </article>
</DocumentFragment>
`;
